src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -563,27 +537,252 @@ internal SafeFileHandle OpenPort(string portName)
                 _handle = null;
                 throw;
             }
+
+            _processReadDelegate = ProcessRead;
+            _processWriteDelegate = ProcessWrite;
+            _ioLoop = Task.Factory.StartNew(
+                IOLoop,
+                CancellationToken.None,
+                TaskCreationOptions.LongRunning,
+                TaskScheduler.Default);
         }
 
-        ~SerialStream()
+        private void FinishPendingIORequests()
         {
-            Dispose(false);
+            while (_readQueue.TryDequeue(out SerialStreamIORequest r))
+            {
+                r.Complete(InternalResources.FileNotOpenException());
+            }
+
+            while (_writeQueue.TryDequeue(out SerialStreamIORequest r))
+            {
+                r.Complete(InternalResources.FileNotOpenException());
+            }
         }
 
         protected override void Dispose(bool disposing)
         {
-            // Signal the other side that we're closing.  Should do regardless of whether we've called
-            // Close() or not Dispose()
-            if (_handle != null && !_handle.IsInvalid)
+            if (disposing)
             {
-                   Interop.Sys.Shutdown(_handle, SocketShutdown.Both);
-                   _handle.Close();
+                _ioLoopFinished = true;
```

Comment:
What happens if someone starts using the stream and then just drops the reference to it?  Is there a finalizer that'll stop the loop, or will it just sit there forever?  If you add a finalizer, should this _ioLoopFinished setting be moved to outside of the disposed check?

Reply:
What's the best way to solve this? If I kill the loop when it's idle that should be fixed - or do we prefer to make the loop static perhaps and just have a WeakReference to `this`?

Currently SerialPort does not call Dispose when called from the finalizer (perhaps something we should change).

Revisit Dispose is already on my TODO list to explore (in the issue description)

Reply:
ok per offline conversation added pausing a loop when it is idle which will solve this issue